lib/dekorator.rb

```diff
@@ -22,9 +21,8 @@ class << self
       def decorate(object_or_collection, with: nil)
         return object_or_collection if decorable_object?(object_or_collection)
 
-        with = _guess_decorator(object_or_collection) if with.nil? && object_or_collection.present?
-
-        raise DecoratorNotFound, "Can't guess decorator for #{object_or_collection.class.name} object" if with.nil?
+        with ||= self if with != :__guess__ && self != Dekorator::Base
+        with = _guess_decorator(object_or_collection) if with.nil? || with == :__guess__
```

Comment:
I made improvements here to make sure that if we use `UserDecorator.decorate(user)` then we use `UserDecorator` and we don't use `_guess_decorator` that consume a bit.

I used `:__guess__` sometimes because it happen that we really want it. For example, if we use `decorate` inside a decorator.